scripts/error_codes.py

```diff
@@ -378,7 +378,10 @@ def main(argv):
         [".h", ".cpp"]
     )
     source_id_to_file_names = find_ids_in_source_files(source_file_names)
-
+    # Transient storage warning which appears twice in the source code
+    # TODO: Remove if we decide the warning is not needed anymore
+    assert(len(source_id_to_file_names["2394"]) == 2)
```

Comment:
The warning could appears more than two times, no? What is the purpose of this?

Reply:
This is regarding the error code appearing in the c++ souce code of the compiler. Currently, the script doesn't allow an error code to be written more than once. This workaround is specific to the case of the transient storage warning and if we remove the warning in the future, this assert should ensure that we also revert this workaround.

Reply:
Oh, right. Now I got it, thanks for clarifying it.

Reply:
We could make it more general by adding functionality in `ErrorReporter` for limiting the number of accepted errors of a specific type. We already have something pretty similar: [`checkForExcessiveErrors()`](https://github.com/ethereum/solidity/blob/v0.8.24/liblangutil/ErrorReporter.cpp#L63-L118) ensures that the overall number of added errors/warnings/infos does not exceed a predefined limit.

With such a mechanism in place we could then have a special header defining error codes that should be limited this way. And we could just exclude that header instead of expecting more than one occurrence in `error_codes.py`.

But that's just FYI. The way you did it is simple, works and we only need it for one specific warning so it's good enough. We could consider making it more generic in the future, but I'd not do it in this PR.

Reply:
Yeah, I thought about this approach of adding a method in `ErrorReporter`, but since it was for a specific case, I took the simpler approach.
I will create an issue about adding this to `ErrorReporter`.